Modules/socketmodule.c

```diff
@@ -7338,1394 +7338,1427 @@ static struct PyModuleDef socketmodule = {
 PyMODINIT_FUNC
 PyInit__socket(void)
 {
-    PyObject *m, *has_ipv6;
+    PyObject *m = NULL;
 
-    if (!os_init())
-        return NULL;
+    if (!os_init()) {
+        goto error;
+    }
 
     Py_SET_TYPE(&sock_type, &PyType_Type);
     m = PyModule_Create(&socketmodule);
-    if (m == NULL)
-        return NULL;
+    if (m == NULL) {
+        goto error;
+    }
 
-    PyModule_AddObject(m, "error", Py_NewRef(PyExc_OSError));
-    socket_herror = PyErr_NewException("socket.herror",
-                                       PyExc_OSError, NULL);
-    if (socket_herror == NULL)
-        return NULL;
-    PyModule_AddObject(m, "herror", Py_NewRef(socket_herror));
-    socket_gaierror = PyErr_NewException("socket.gaierror", PyExc_OSError,
-        NULL);
-    if (socket_gaierror == NULL)
-        return NULL;
-    PyModule_AddObject(m, "gaierror", Py_NewRef(socket_gaierror));
-    PyModule_AddObjectRef(m, "timeout", PyExc_TimeoutError);
+#define ADD_EXC(MOD, NAME, VAR) do {                                \
```

Comment:
nit: could call this ADD_OSERROR_EXC?

Reply:
Yeah, I thought about making the naming more explicit. I'll consider your suggestion before merging 🙂 Thanks for the review!

Reply:
I added base as an argument to the macro; it is better to be explicit.